src/mongocxx/pool.cpp

```diff
@@ -33,6 +33,18 @@
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 
+// Attempts to create a new client pool using the uri. Throws an exception upon error.
+static mongoc_client_pool_t* construct_client_pool(mongoc_uri_t* uri) {
+    bson_error_t error;
+    auto pool = libmongoc::client_pool_new_with_error(uri, &error);
+    if (error.code) {
```

Comment:
I'm checking by error code here because the `interpose` function for `client_pool_new_with_error` in the `a pool is created with the correct MongoDB URI` test returns nullptr to the pool. Because of this, writing the if statement as `if (!pool)` won't work. I'm unsure on if returning an empty `mongoc_client_pool_t *` from the interpose is the correct solution here but I understand why checking `error.code` isn't intuitive.

Reply:
Checking `if (error.code)` may work, but I expect this may require `bson_error_t error = {0};` to ensure `error` is initialized when no error occurs. `mongoc_client_pool_new_with_error` may not zero `error` if no error occurs.

I suggest checking `if (!pool)` and updating the test mocks to not use `interpose` to replace the C driver function call. `visit` can be used to call a function in addition to the C driver function.

Here is a suggested commit with a description: https://github.com/mongodb/mongo-cxx-driver/commit/be69c679ece863011eb0d2a2012c0bee6e0dcc65